vscode/src/extension.ts

```diff
@@ -415,9 +415,14 @@ export function activate(context: ExtensionContext): VSNetBeansAPI {
                 return;
             }
 
-            // first give the context, then the open-file hint in the case the context is not specific enough
-            const res = await vscode.commands.executeCommand(COMMAND_PREFIX + '.new.from.template', contextUri(ctx)?.toString(), vscode.window.activeTextEditor?.document?.uri?.toString());
-
+            // first give the template (if present), then the context, and then the open-file hint in the case the context is not specific enough
+            const params = [];
+            if (typeof template === 'string') {
+                params.push(template);
+            }
+            params.push(contextUri(ctx)?.toString(), vscode.window.activeTextEditor?.document?.uri?.toString());
+            const res = await vscode.commands.executeCommand(COMMAND_PREFIX + '.new.from.template', ...params);
+            
             if (typeof res === 'string') {
                 let newFile = vscode.Uri.parse(res as string);
                 await vscode.window.showTextDocument(newFile, { preview: false });
```

Comment:
Do incorporate the fix in the function const `runDebug` on line 642 where:
```
            if (!methodName) {
                debugConfig['methodName'] = methodName;
            }
```
is fixed to:
```
            if (methodName) {
                debugConfig['methodName'] = methodName;
            }
```